overlord/snapstate/export_test.go

```diff
@@ -193,3 +195,44 @@ func ByKindOrder(snaps ...*snap.Info) []*snap.Info {
 	sort.Sort(byKind(snaps))
 	return snaps
 }
+
+func MockModelWithBase(baseName string) func() {
```

Comment:
As a minor, I'd suggest preserving the convention of naming the return function explicitly as _(restore func())_. We know the convention by heart now, but it's still nicer to see the name there and be sure to know what it is, even more for passers by.